cmd/apiserver/app/server/etcd_options.go

```diff
@@ -32,10 +31,18 @@ type EtcdOptions struct {
 	*genericserveroptions.EtcdOptions
 }
 
+const (
+	// DefaultEtcdPathPrefix is the default prefix that is prepended to all
+	// resource paths in etcd.  It is intended to allow an operator to
+	// differentiate the storage of different API servers from one another in
```

Comment:
If it is intended to differentiate API servers, shouldn't the prefix be `/servicecatalog` instead? Or do you live it for option overriding?

Reply:
Defaults should behave coherently if you made all api servers for a single cluster use the same etcd.

Reply:
@liggitt right, so what happens if two API servers will define the same `/registry` prefix (sharing the same etcd)? Isn't that what this option tries to avoid? Shouldn't this option be API server specific to avoid clashing prefixes?

I probably misunderstand the mechanics of what this option tries to solve.

Reply:
etcd prefix + api group prefix + resource prefix

* etcd prefix defaults to /registry
* API group prefix should be the group name (the kube api server is all over the map here for historic reasons… new api groups behave well)

This means a cluster admin can start multiple extension servers and with their defaults, they play nicely within the /registry prefix

If an admin wanted to colocate multiple clusters in a single etcd (which sounds nuts to me, but oh well…), they'd start the servers (kube-apiserver + any addon servers) for a particular cluster with an identical prefix (e.g. /cluster1 instead of /registry)

Reply:
thanks for the clarification